pkg/operator/status.go

```diff
@@ -0,0 +1,151 @@
+// there are no replica instances remaining from the previous deployment.
+// There may still be additional replica instances being created.
+func isDeploymentStatusAvailableAndUpdated(deploy *appsv1.Deployment) bool {
+	return deploy.Status.AvailableReplicas > 0 &&
+		deploy.Status.ObservedGeneration >= deploy.Generation &&
+		deploy.Status.UpdatedReplicas == deploy.Status.Replicas
+}
+
+func isDeploymentStatusComplete(deploy *appsv1.Deployment) bool {
+	replicas := int32(1)
+	if deploy.Spec.Replicas != nil {
+		replicas = *(deploy.Spec.Replicas)
+	}
+	return deploy.Status.UpdatedReplicas == replicas &&
+		deploy.Status.Replicas == replicas &&
+		deploy.Status.AvailableReplicas == replicas &&
+		deploy.Status.ObservedGeneration >= deploy.Generation
+}
+
+func (c *serviceCAOperator) syncStatus(operatorConfigCopy *operatorv1.ServiceCA, deployments []string) (bool, error) {
+	ready := 0
+	existingDeploymentsAndReplicas := 0
+	var statusErr error
+	for _, dep := range deployments {
+		existing, err := c.appsv1Client.Deployments(operatorclient.TargetNamespace).Get(dep, metav1.GetOptions{})
+		if apierrors.IsNotFound(err) {
+			glog.Infof("Deployment %s does not exist", dep)
+			statusMsg := fmt.Sprintf("Deployment %s does not exist", dep)
+			c.setFailingStatus(operatorConfigCopy, "Error getting service-ca-operator managed deployments", statusMsg)
+			return false, statusErr
+		} else if existing.DeletionTimestamp != nil {
+			glog.Infof("Deployment %s is being deleted", dep)
+			statusMsg := fmt.Sprintf("Deployment %s is being deleted", dep)
+			c.setFailingStatus(operatorConfigCopy, "Deletion timestamp found in service-ca-operator managed deployments", statusMsg)
+			return false, statusErr
+		} else if !isDeploymentStatusAvailable(existing) {
+			glog.Infof("Deployment %s does not have available replicas", dep)
+			statusMsg := fmt.Sprintf("Deployment %s does not have available replicas", dep)
+			c.setProgressingStatus(operatorConfigCopy, statusMsg)
+			return false, statusErr
+		} else if err != nil {
+			c.setFailingStatus(operatorConfigCopy, "Error getting service-ca-operator managed deployments", err.Error())
+			return false, statusErr
+		} else {
+			existingDeploymentsAndReplicas++
+		}
+		if !isDeploymentStatusComplete(existing) {
+			glog.Infof("The deployment %s has not completed", dep)
+			statusMsg := fmt.Sprintf("Deployment %s has not completed", dep)
+			c.setProgressingStatus(operatorConfigCopy, statusMsg)
+			return false, statusErr
+		} else {
+			if isDeploymentStatusAvailableAndUpdated(existing) {
+				glog.Infof("Deployment %s is available and updated", dep)
+				ready++
+			}
+		}
+	}
+	// only report version if all deployments are available and updated
+	if ready == len(deployments) {
+		glog.Infof("All deployments managed by service-ca-operator are available and updated")
+		c.setAvailableStatus(operatorConfigCopy)
+		return true, statusErr
+		// set Available if replica instances are created.  Don't report new version if
```

Comment:
This comment should be inside the condition branch it applies to.

Reply:
fixed